xray/segment.go

```diff
@@ -61,6 +62,13 @@ func BeginSegment(ctx context.Context, name string) (context.Context, *Segment)
 }
 
 func BeginSegmentWithSampling(ctx context.Context, name string, r *http.Request, traceHeader *header.Header) (context.Context, *Segment) {
+	// If SDK is disabled then return with an empty segment
+	sdkDisable := Disabled()
+	if sdkDisable {
+		seg := &Segment{}
```

Comment:
As an aside, it would be less fragile if we returned a no-op recorder when the SDK is disabled (wouldn't be flippable at runtime but I don't think that's needed). A bigger change so can think about it for later.

Reply:
good point!